gateway/src/main/java/ru/practicum/shareit/client/BaseClient.java

```diff
@@ -0,0 +1,121 @@
+package ru.practicum.shareit.client;
+
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.lang.Nullable;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestTemplate;
+
+public class BaseClient {
+    protected final RestTemplate rest;
+
+    public BaseClient(RestTemplate rest) {
+        this.rest = rest;
+    }
+
+    protected ResponseEntity<Object> get(String path) {
+        return get(path, null, null);
+    }
+
+    protected ResponseEntity<Object> get(String path, long userId) {
+        return get(path, userId, null);
+    }
+
+    protected ResponseEntity<Object> get(String path, Long userId, @Nullable Map<String, Object> parameters) {
+        return makeAndSendRequest(HttpMethod.GET, path, userId, parameters, null);
+    }
+
+    protected <T> ResponseEntity<Object> post(String path, T body) {
```

Comment:
Если параметр во всех реализациях всегда одинаковый, как например `path = ""`, можно его схлопнуть внутрь метода:
```suggestion
    protected <T> ResponseEntity<Object> post(T body) {
         return post("", null, null, body);
    }
```
Ниже есть несколько похожих случаев, можно тоже поправить

Reply:
Не во всех методах классов которые его наследуют используется путь "". Например в итем контроллере. 
Прошу оставить такую реализацию.